src/ragas/evaluation.py

```diff
@@ -240,17 +248,32 @@ def evaluate(
             callbacks=evaluation_group_cm,
         )
         row_run_managers.append((row_rm, row_group_cm))
-        _ = [
-            executor.submit(
-                metric.single_turn_ascore,
-                sample,
-                row_group_cm,
-                name=f"{metric.name}-{i}",
-                timeout=run_config.timeout,
-            )
-            for metric in metrics
-            if isinstance(metric, SingleTurnMetric)
-        ]
+        if sample_type == SingleTurnSample:
+            _ = [
+                executor.submit(
+                    metric.single_turn_ascore,
+                    sample,
+                    row_group_cm,
+                    name=f"{metric.name}-{i}",
+                    timeout=run_config.timeout,
+                )
+                for metric in metrics
+                if isinstance(metric, SingleTurnMetric)
```

Comment:
what happens if it is not `SingleTurnMetric` it will fail silently confusing the user. It is better to do this in the validation step ealier and then we can safely assume it is SingleTurn

Reply:
That validation function is this one
https://github.com/explodinggradients/ragas/blob/fe684673a9b79d11ea1e3fe04c18a27696788146/src/ragas/evaluation.py#L168